src/document/actions/index.ts

```diff
@@ -6,96 +7,121 @@ import {
     RedoAction,
     State,
     UndoAction,
+    UndoRedoProcessResult,
 } from '../types';
 import { hashDocument, replayOperations } from '../utils/base';
-import { loadState } from './creators';
+import { loadState, noop } from './creators';
 
 // updates the name of the document
 export function setNameOperation<T>(document: T, name: string): T {
     return { ...document, name };
 }
 
-// undoes the last `count` operations
+
 export function undoOperation<T, A extends Action, L>(
     document: Document<T, A, L>,
     action: UndoAction,
-    wrappedReducer: ImmutableStateReducer<T, A, L>,
-): Document<T, A, L> {
-    const { scope, input: count } = action;
-    const revision = document.revision[action.scope];
-
-    // undo can't be higher than the number of active operations
-    const undoCount = Math.min(count, revision);
-
-    // builds the global state from the initial data without the
-    // undone operations
-    const operations = document.operations[scope].slice(
-        0,
-        revision - undoCount,
-    );
-    const newDocument = replayOperations(
-        document.initialState,
-        {
-            ...document.operations,
-            [scope]: operations,
-        },
-        wrappedReducer,
-    );
-
-    // updates the state and the revision number but
-    // keeps the operations history to allow REDO
-    return {
-        ...newDocument,
-        operations: document.operations,
-        revision: {
-            ...document.revision,
-            [scope]: document.revision[scope] - undoCount,
-        },
+    skip: number,
+): UndoRedoProcessResult<T, A, L> {
+    const scope = action.scope;
```

Comment:
If you do `const { scope, input } = action;` here then you can use the `input` var inside the `produce` function, instead of `draft.action as UndoAction`

Reply:
fixed in [b39596d](https://github.com/powerhouse-inc/document-model/pull/29/commits/b39596da1c0fad4c742170a6c43134c4799d830f)